app/controllers/enterprises/YouHaveNotAddedAnyAssociatedEnterprisesController.scala

```diff
@@ -40,25 +40,37 @@ class YouHaveNotAddedAnyAssociatedEnterprisesController @Inject()(
     navigator: NavigatorForEnterprises,
     identify: IdentifierAction,
     getData: DataRetrievalAction,
+    requireData: DataRequiredAction,
     formProvider: YouHaveNotAddedAnyAssociatedEnterprisesFormProvider,
     val controllerComponents: MessagesControllerComponents,
     renderer: Renderer
 )(implicit ec: ExecutionContext) extends FrontendBaseController with I18nSupport with NunjucksSupport with RoutingSupport {
 
   private val form = formProvider()
 
-  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData).async {
+  def onPageLoad(mode: Mode): Action[AnyContent] = (identify andThen getData andThen requireData).async {
     implicit request =>
 
-      val preparedForm = request.userAnswers.flatMap(_.get(YouHaveNotAddedAnyAssociatedEnterprisesPage)) match {
+      val namesOfAssociatedEnterprises: IndexedSeq[String] = request.userAnswers.get(AssociatedEnterpriseLoopPage) match {
+        case Some(list) =>
+          for {
+            enterprise <- list
+          } yield {
+            enterprise.nameAsString
+          }
```

Comment:
isn't this just list.map(_.nameAsString) ?

Reply:
the whole this is actually get(...).map(_.nameAsString).getOrElse(IndexSeq.empty)